docs/example_usage.md

```diff
@@ -0,0 +1,161 @@
+      }
+    }
+  },
+  "Package": {
+    "Name": "dep1",
+    "VersionTag": "v1.0.0",
+    "PlatformString": {
+      "Mode": "auto"
+    },
+    "IsLibrary": true,
+    "IsDevLib": true,
+    "IsDebug": false
+  },
+  "DockerMatrix": {
+    "ImageNames": [
+      "image1"
+    ]
+  }
+}
+```
+
+All fields of json definitions (Config) are described in Packager documentation. Only important
+fields for our use case will be described further. The `DependsOn` field contains all dependency
+Packages which must be defined in the same Context. The `Git/URI` must be a uri to a CMake based
+git repository with tag specified by `Git/Revision`. In `Build/CMake/Defines` can be specified all
+CMake options. The `Package/Name` field must be same as name of `package_group_name`. And all
+images in `DockerMatrix/ImageNames` must be defined in `docker` directory of same Context.
+
+The debug variation of `dep1` Package will be very similar with changes of `CMAKE_BUILD_TYPE` and
+`Package/IsDebug` flag. The `dep2` Package will be the same only with empty `DependsOn` array.
+
+## Build a Package
+
+After definition of Package `dep1` we can build it with Packager.
+
+If the used image in our Package is not build on our system, it must be build with `build-image`
+Packager command. The command for building `image1` image could be:
+
+```bash
+bap-builder build-image --context /path/to/package/context --image-name image1
+```
+
+The Packager command `build-package` for building Packages requires these parameters:
+
+ - context
+ - image-name
+ - output-dir
+
+The `context` is a path to the Package Context, the `image-name` is a target image for Package and
+`output-dir` is a path to the Package Repository, which already can have built Packages or it can
+be an empty Git repository.
+
+We want Packager to put this Package to
+[BringAuto's main Package Repository](https://gitea.bringauto.com/fleet-protocol/package-repository).
+Firstly, the repository must be cloned, then the path to it can be given to Packager executable.
+
+The command for building Package `dep1` is:
+
+```bash
+bap-builder build-package --context /path/to/package/context --image-name image1 --output-dir /path/to/package/repository --name dep1 --build-deps
+```
+
+This command builds a Package `dep1` defined in Context for `image1` image, creates an archive
+of this Package and copies it to the output-dir (Package Repository). The command with
+`--build-deps` flag also builds all dependencies of given Package. In our case it also builds the
+`dep2` Package. More flags and settings of Packager are described in its documentation.
+
+After the Package is in Package Repository we can push it to origin.
+
+## Include Package in an application
+
+We have built a dependency Package `dep1` and put it to remote BringAuto's Package Repository. Now
+we can use this Package in a CMake based application.
+
+First we need to have cmakelib (link in [Introduction](./index.md)) installed on our system.
```

Comment:
There is almost no information about cmakelib. Maybe it will be nice to add more content info....